### PR TITLE
MM-36600 : Animate emoji skin tone selector

### PR DESCRIPTION
**Pull request by Jio007**
#### Summary
- Adding animation for the emoji skin tone selector

#### Ticket Link
  Fixes https://github.com/mattermost/mattermost-server/issues/18429
  Jira Ticket: https://mattermost.atlassian.net/browse/MM-36600

#### Screenshots
|  before  |  after  |
|----|----|
| before | after |

#### Release Note

```release-note
Animate emoji skin tone selector
```

### CODE REVIEW

**Comment by M-ZubairAhmed** [comment]
@Jio007 this looks great i am going to ask UX review on this one to see if it matches our figma prototype

**Comment by Jio007** [comment]
> @Jio007 this looks great i am going to ask UX review on this one to see if it matches our figma prototype
> 
> https://www.figma.com/file/GPTxUI7ACBbbdRfM8Dbtl6/MM-25435-Emoji-Picker-UI-Refresh?node-id=10%3A1220

Thanks for taking a look @M-ZubairAhmed 👍🏾

**Comment by M-ZubairAhmed** [comment]
@matthewbirtch cloud instance is now created

**Comment by nevyangelova** [comment]
Thanks @Jio007 looking at the prototype, their animation seems to be a tad faster, but i do prefer the version we have here personally. All looks good to me. Waiting for UX review.

**Review by M-ZubairAhmed — APPROVED** [review]

File: sass/components/_emoticons.scss [hunk 1]
```diff
@@ -268,13 +268,35 @@
...
+
+        &-exit {
+            position: absolute;
+            width: 100% !important;
+            transform: translateX(0%);
+        }
+
+        &-exit-active {
+            transform: translateX(90%);
+            transition: transform 300ms ease-out;
```
Comment (matthewbirtch) [inline sass/components/_emoticons.scss:289]: can we tweak this to:
```
transition: transform 200ms ease-in-out;
```

File: sass/components/_emoticons.scss [hunk 2]
```diff
@@ -268,13 +268,35 @@
...
     margin: 12px 12px 0 12px;
 
+    .skin-tones-animation {
+        &-enter {
+            transform: translateX(100%);
+        }
+
+        &-enter-active {
+            transform: translateX(0%);
+            transition: transform 300ms ease-out;
```
Comment (matthewbirtch) [inline sass/components/_emoticons.scss:278]: can we tweak this to:
```
transition: transform 200ms ease-in-out;
```

**Review by matthewbirtch — CHANGES_REQUESTED** [review]
Just a couple minor tweaks to the animation. Other than that, we're good!

**Comment by Jio007** [comment]
@matthewbirtch @nevyangelova thanks for the review.

**Review by matthewbirtch — APPROVED** [review]
Nice and smooth! Thanks for this @Jio007!

File: sass/components/_emoticons.scss [hunk 3]
```diff
@@ -268,13 +268,35 @@
...
+        }
+
+        &-enter-active {
+            transform: translateX(0%);
+            transition: transform 200ms ease-in-out;
+        }
+
+        &-exit {
+            position: absolute;
+            width: 100% !important;
```
Comment (nevyangelova) [inline sass/components/_emoticons.scss:283]: if you add the width prop in the other steps you won't need to use the !important tag. We are trying to stay away from using it

**Review by nevyangelova — CHANGES_REQUESTED** [review]

File: sass/components/_emoticons.scss (hunk 3 above)
Comment (Jio007) [reply sass/components/_emoticons.scss:283]: got it, I will take a look

**Review by nevyangelova — APPROVED** [review]
LGTM

**Review by jgilliam17 — APPROVED** [review]
Thanks @Jio007 
Tested, looks good to merge.
Verified emoji skin tone selector animation.
[E2E report](https://mattermost-cypress-report.s3.amazonaws.com/245605-a8eab67-onprem-ent-webapp-pr-11114-mm-ee-test:a8eab67/mochawesome.html)- no PR related failures. 
@M-ZubairAhmed can you please approve the last workflow check and merge - thanks 🙂

**Pull request merged**